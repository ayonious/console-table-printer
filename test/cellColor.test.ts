import chalk from 'chalk';
import { Table } from '../index';

describe('Example: Print a simple Table with cell colors', () => {
  it('cell colors are working: ++++Might Fail on CI++++++++', () => {
    // Create a table
    const p = new Table();

    // add rows with color
    p.addRow({ index: 1, text: chalk.red('red'), value: 10.212 });
    p.addRow({ index: 2, text: chalk.green('green'), value: 20.0 });
    p.addRows([
      // adding multiple rows are possible
      { index: 3, text: chalk.yellow('yellow'), value: 100 },
      { index: 4, text: chalk.bgMagenta('magenta'), value: 300 },
    ]);

    // print
    p.printTable();
    expect(p.render()).toMatchSnapshot();
  });
});
