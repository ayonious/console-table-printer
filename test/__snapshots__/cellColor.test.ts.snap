// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Example: Print a simple Table with cell colors cell colors are working 1`] = `
"┌───────┬─────────┬────────┐
│[37m [0m[01mindex[0m │[37m [0m[01m   text[0m │[37m [0m[01m value[0m │
├───────┼─────────┼────────┤
│[37m [0m[37m    1[0m │[37m [0m[37m    [31mred[39m[0m │[37m [0m[37m10.212[0m │
│[37m [0m[37m    2[0m │[37m [0m[37m  [32mgreen[39m[0m │[37m [0m[37m    20[0m │
│[37m [0m[37m    3[0m │[37m [0m[37m [33myellow[39m[0m │[37m [0m[37m   100[0m │
│[37m [0m[37m    4[0m │[37m [0m[37m[45mmagenta[49m[0m │[37m [0m[37m   300[0m │
└───────┴─────────┴────────┘"
`;
